### PR TITLE
Outvec stream rebased

### PR DESCRIPTION
**Pull request by orenc17**
This PR brings support to the new changes in psa_write and pas_call
the user may now send up to 3 respose buffers in an iovec structure

Tests have been added acordingly

This replaces PR #1

### CODE REVIEW

File: spm/spm_server.c [hunk 1]
```diff
@@ -211,29 +222,20 @@ void psa_write(psa_handle_t msg_handle, size_t offset, const void *buffer, size_
...
-            active_iovec->iov_len
-            );
+
+    if (!is_buffer_accessible(active_iovec->base, num_bytes)) {
+        SPM_PANIC("buffer is inaccessible\n");
     }
 
-    memcpy((uint8_t *)(active_iovec->iov_base) + offset, buffer, bytes);
+    memcpy((uint8_t *)(active_iovec->base), buffer, num_bytes);
+    active_iovec->base = (void *)((uint8_t*)active_iovec->base + num_bytes);
```
Comment (danny4478) [inline spm/spm_server.c]: add space between uint8_t and *

[the remark this answers is not in the document]
Comment (orenc17A) [reply spm/spm_server.c]: please elaborate

[the remark this answers is not in the document]
Comment (danny4478) [reply spm/spm_server.c]: if (active_iovec->base> UINT32_MAX - num_bytes) then panic

[the remark this answers is not in the document]
Comment (orenc17) [reply spm/spm_server.c]: wrap around is checked in `is_buffer_accessible`
in `psa_write`:
* the input buffer is checked at line 209
* the output buffer is checked at line 231

so all the bases are covered

[the remark this answers is not in the document]
Comment (danny4478) [reply spm/spm_server.c]: Good!

**Pull request merged**